sdks/java/testing/nexmark/build.gradle

```diff
@@ -98,7 +103,16 @@ if (shouldProvideSpark) {
 //   -Pnexmark.args
 //       Specify the command line for invoking org.apache.beam.sdk.nexmark.Main
 task run(type: JavaExec) {
+  dependsOn ":beam-runners-google-cloud-dataflow-java-legacy-worker:shadowJar"
```

Comment:
This should only be done if the Dataflow runner is being used. Note that nexmark is being used for many runners.

See `shouldProvideSpark`, you could copy the idea and add a `isDataflow` variable.